solutions/cses-1139.mdx

```diff
@@ -0,0 +1,113 @@
+---
+id: cses-1139
+source: CSES
+title: Distinct Colors
+author: Timothy Gao
+---
+
+## Editorial
+
+Let us consider the Euler Tour of the tree. We flatten the tree into an array, where each node corresponds to a range in this array. Now, we are essentially tasked the find the number of distinct values in a range $N$ times(once for each node).
+
+There are a variety of ways to approach the problem from here.
+
+**Solution: BIT**
+
+One such solution is with a binary indexed tree. Let us consider iterating through the Euler tour array from left to right. 
+When we consider a node at index $i$ in the Euler tour array, its subtree range will be $[j, i]$, where $j<=i$.
```

Comment:
Use \leq instead of <=. Also, what is j?

Reply:
I'm not that familiar with latex how do I use that?

Reply:
$j\leq i$